src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -10,13 +10,56 @@
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.FindSymbols.Finders;
 using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Shared.Utilities;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
     using ProjectToDocumentMap = Dictionary<Project, MultiDictionary<Document, (SymbolAndProjectId symbolAndProjectId, IReferenceFinder finder)>>;
 
+    internal class FindReferencesSearchOptions
```

Comment:
>FindReferencesSearchOptions [](start = 19, length = 27)

I had trouble finding this type. Probably should be in its own file.

Since this only stores a bit of information, maybe an enum/struct type would be appropriate.

Reply:
> I had trouble finding this type. Probably should be in its own file.

Agreed.  Will do.

> Since this only stores a bit of information, maybe an enum/struct type would be appropriate.

In general, IDE opts toward more future-proofing here, with rich objects so we don't box ourselves in in the future.

Reply:
Since this isn't a public API, I'm not sure I see the future-proofing benefits. But I don't feel strongly on this :-)

---
In reply to: [205585178](https://github.com/dotnet/roslyn/pull/28231#discussion_r205585178) [](ancestors = 205585178)

Reply:
That's fair.  Though, in a sense, this helps serve as an advisory shape for how we expect options to be added here.  i.e. if it was an enum, that would be strongly implying that we wanted all future options to be limited to a numeric value.  If we then needed something more complex in the future that could lead to confusion.

This doesn't seem problematic *if* we haven't made anything public.  However, it's quite possible thta a year from now we go and make these guys public.  If it's an enum, and we didn't think about it enough, then we can ship tha way and box ourselves into a corner (it's happened before :)).  So this is the somewhat consistent, somewhat future-proofed way to do options here. :)